src/visualizers/panels/ExecutionIndex/ExecutionIndexControl.js

```diff
@@ -38,45 +41,88 @@ define([
     };
 
     ExecutionIndexControl.prototype._initWidgetEventHandlers = function () {
-        this._widget.setExecutionDisplayed = this.setExecutionDisplayed.bind(this);
+        this._widget.setDisplayedExecutions = this.setDisplayedExecutions.bind(this);
     };
 
-    ExecutionIndexControl.prototype.setExecutionDisplayed = function (id, bool) {
-        var lines = this._linesForExecution[id] || [],
-            otherLines,
-            wasMultiLine = this.displayedExecCount() > 1,
-            isMultiLine;
+    ExecutionIndexControl.prototype.setDisplayedExecutions = function (beforeClickIds, afterClickIds) {
+        let addedExecutions = afterClickIds.filter(id => !beforeClickIds.includes(id));
+        let removedExecutions = beforeClickIds.filter(id => !afterClickIds.includes(id));
+        let added = addedExecutions.length > 0;
+        let updatedExecutions = added ? addedExecutions : removedExecutions;
+        this._updateGraphData(...updatedExecutions, added);
```

Comment:
It appears that the spread operator is used in place of `[0]` here and works only because a single execution is updated at a time. If more than one execution could be removed at once, this would fail (as the second id would be passed as the `bool` argument and is truthy).

Reply:
Yeah, This is true, but is the only behavior. Since, this method is called on a checklist click. I thought, at most one execution gets removed or added at a time, which seems reasonable to be.

Reply:
This method has been refactored not to use the spread operator.